curriculum/challenges/english/25-front-end-development/lecture-working-with-relative-and-absolute-units/672bb851b068e3954a05b9b1.md

```diff
@@ -8,7 +8,130 @@ dashedName: what-is-the-calc-function
+
+For this, you pass something called an expression as an argument. An expression is a combination of values and operators that produces a result.
+
+This is how you can call the `calc()` function. You write the name calc, followed by parentheses, and within the parentheses, you write the expression:
+
+```css
+calc(expression)
+```
+
+The expression is evaluated to calculate the final result. "Evaluated" just means that the values and operators are converted into a single value behind the scenes. The result is assigned to the CSS property where the calculation is being made.
+
+You can perform calculations on values that represent length, angle, time, percentages, numbers, and colors. You can also combine different units like pixels, percentages, and ems.
+
+With numbers, all the values in the expression, also called the operands, must have their corresponding units, like `px`, `em`, and percentage (`%`). Depending on the operator, different operands may have different units.
+
+You can use the addition (`+`), subtraction (`-`), multiplication (`*`), and division (`/`) operators in the expression.
+
+If there are multiple operands and operators, `calc()` will follow the standard operator precedence rule. You can also add parentheses to establish the order of the operations if needed.
+
+In the example below, you can see a `div` with the text `Hello, World!`:
+
+```html
+<div>Hello, World!</div>
+```
+
+Using the CSS type selector for selecting the `div`, you can style it with white text and a dark blue background: 
+
+```css
+div {
+  color: white;
+  background-color: #1b1b32;
+  width: calc(50% - 20px);
+}
+```
+
+What's new here is that the width is calculated dynamically. Notice how we are calling the `calc()` function and passing an expression as an argument. The expression has two operands with different units and one operator, the subtraction operator.
+
+Percentage is a relative unit. The value (`50%`) will be determined by the width of the parent container. Then, `20px` is subtracted from the value. The result of this expression will determine the width of the `div`.
+
+The width of the `div` is approximately half the total width of its container, and if you resize the parent container, the width will be recalculated automatically.
+
+That's the key advantage of using `calc()`. You can determine the value of a CSS property dynamically based on different aspects of the application or viewport.
+
+The expression can also contain CSS functions and variables if you need to use them in your calculations. You'll learn more about CSS variables in the next lectures.
+
+Great. Now that you know about the basics of the `calc()` function, let's cover some of its best practices.
+
+First, you must surround the addition (`+`) and subtraction (`-`) operators with whitespace.
+
+For example, this would not be a valid expression because the subtraction operator is immediately before the second operand.
```

Comment:
```suggestion
For example, the expression below would not be a valid expression because the subtraction operator is immediately before the second operand.
```